### PR TITLE
Use a spinner on team creation, move to a WaitingKey

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers

### CODE REVIEW

**Comment by cjb** [comment]
@chrisnojima Thanks, done!

@zanderz Sure, sounds good.  (Though, oftentimes we're using createIncrementWaiting because there's more than one RPC to wait on, or some other kind of chain of events that we want to hang out until the end of.)

**Comment by cjb** [comment]
(I suppose given the above, it makes more conceptual sense to tie *sagas* to waiting keys rather than RPCs.  Tying RPCs only works well if there's one of them and there's nothing else to do afterwards before returning to the user.  If we just tie the RPC, then the user's spinner might return before the app is ready for them.)

**Comment by chrisnojima** [comment]
I'd argue the waiting key should be tied to a concept. Something reflected in the ui. I'd multiple rpcs are required to fulfill it they can both wait on the same key (it's a ref count)

**Comment by cjb** [comment]
@chrisnojima bump, made the waitingkey change

File: shared/actions/teams.js [hunk 1]
```diff
@@ -438,7 +435,7 @@ const _createNewTeamFromConversation = function*(
 
   if (participants) {
     yield Saga.put(TeamsGen.createSetTeamCreationError({error: ''}))
-    yield Saga.put(TeamsGen.createSetTeamCreationPending({pending: true}))
+    yield Saga.put(WaitingGen.createIncrementWaiting({key: Constants.teamCreationWaitingKey}))
```
Comment (chrisnojima) [inline shared/actions/teams.js]: why not here also

Comment (chrisnojima) [reply shared/actions/teams.js]: did you actually try it with the same key to those other calls, we can easily make a grace period for the spinners to not stop immediately

Comment (cjb) [reply shared/actions/teams.js]: @chrisnojima I didn't notice a visual glitch, though the render function does get called with waiting=false inbetween teamCreate and teamAdd after your change.  (Maybe the return from render() only makes it to a frame on the screen on slower hardware?)

Adding a grace period doesn't feel great to me -- feels like we're just trying to emulate what the code is already doing, but in a way that doesn't have any visibility into what the user's actually waiting on, we could chat more about what we think is the right granularity to spin on.

Comment (chrisnojima) [reply shared/actions/teams.js]: i think if it spins, then stops for a couple of frames (doubtful) and spins some more thats fine. we have plenty of worse situations in the app (no spinners at all) so i think this is a case we can focus on way down the line

Comment (cjb) [reply shared/actions/teams.js]: @chrisnojima Okay, done.

**Pull request merged**